src/rt/sched/cown.h

```diff
@@ -535,42 +535,40 @@ namespace verona::rt
      **/
     static void fast_send(MultiMessage::MultiMessageBody* body, EpochMark epoch)
     {
-      size_t count = body->count;
-      Cown** cowns = body->cowns;
-
-      Alloc* alloc = ThreadAlloc::get();
-      assert(body->index < count);
-      size_t last = count - 1;
+      auto* alloc = ThreadAlloc::get();
+      const auto last = body->count - 1;
+      assert(body->index <= last);
 
-      backpressure_ensure_progress(body);
+      const auto make_unmutable = backpressure_ensure_progress(body);
 
-      for (; body->index < count; body->index++)
+      for (; body->index < body->count; body->index++)
       {
         MultiMessage* m = MultiMessage::make_message(alloc, body, epoch);
-        Systematic::cout() << "MultiMessage " << m << " index " << body->index
-                           << " fast requesting " << cowns[body->index]
-                           << std::endl;
+        auto* next = body->cowns[body->index];
+        Systematic::cout() << "MultiMessage " << m << ": fast requesting "
+                           << next << ", index " << body->index << std::endl;
 
-        bool needs_scheduling =
-          cowns[body->index]->send<YesTransfer, YesTryFast>(m);
+        bool needs_scheduling = next->send<YesTransfer, YesTryFast>(m);
         if (!needs_scheduling)
         {
           // Case 1: target cown was already scheduled.
+          // The message may have beeen freed at this point.
           Systematic::cout()
-            << "MultiMessage " << m << " fast send interrupted" << std::endl;
+            << "MultiMessage " << m << ": fast send interrupted" << std::endl;
+          if (make_unmutable)
+            next->unmute(true);
```

Comment:
I'm worried `next` could have been deallocated here.

Reply:
Holding an epoch from 551 -> 559 should fix the issue.

Reply:
fixed